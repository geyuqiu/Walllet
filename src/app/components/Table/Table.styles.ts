export const tableStyle = `
	table {
		margin: 0;
		padding: 0;
		width: 100%;
		border: none;
		border-collapse: collapse;

		thead {
			th.hasBorder:not(:last-child):after {
				content: "";
		    height: 50%;
		    position: absolute;
		    right: 0;
		    top: 25%;
		    border-left-width: 1px;
		    border-color: inherit;
			}
			
			.first\\:pl-0:first-child {
		    padding-left: 11px !important;
			}
			
			.last\\:pr-0:last-child {
		    padding-right: 11px !important;
			}
		}
	}
`;
