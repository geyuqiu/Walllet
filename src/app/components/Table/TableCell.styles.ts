import tw from "twin.macro";

const baseStyle = tw`flex items-center md:p-5 p-5 sm:p-0 my-1 transition-colors duration-100`;

const getHeight = () => tw`h-6`;

const getVariant = (variant: "start" | "middle" | "end"): any => {
	if (variant === "start") {
		return tw`pl-4 -ml-4 rounded-l-lg`;
	}

	if (variant === "end") {
		return tw`pr-4 -mr-4 rounded-r-lg`;
	}
};

export const getStyles = ({ variant }: any) => [baseStyle, getHeight(), getVariant(variant)];
