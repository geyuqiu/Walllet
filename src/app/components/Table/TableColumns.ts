export const timestampAccessor = 'timestamp';

export const tableColumns = [
	{
		Header: 'Txid',
		accessor: 'id',
	},
	{
		Header: 'Sender',
		accessor: 'sender',
	},
	{
		Header: 'Recipient',
		accessor: 'recipient',
	},
	{
		Header: 'Timestamp',
		accessor: timestampAccessor,
		className: "justify-center"
	},
	{
		Header: 'Amount',
		accessor: 'amount',
		className: "justify-end"
	},
	{
		Header: 'Fee',
		accessor: 'fee',
		className: "justify-end"
	}
];
