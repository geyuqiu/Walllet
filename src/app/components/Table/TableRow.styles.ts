import tw, {css} from "twin.macro";

import {TableRowFunction} from "./TableRow";

const baseStyle = tw`transition-colors duration-100`;

const getCursorStyles = (onClick?: TableRowFunction): any => (onClick ? tw`cursor-pointer` : "");

const getBorderStyles = (border?: boolean): any =>
	border ? tw`border-b last:border-b-0 border-solid border-theme-secondary-300` : "";

const getHoverStyles = (isSelected?: boolean): any =>
	css`
		@media
		  only screen 
	    and (min-width: 767px) {
			&:hover td {
				${isSelected
					? tw`bg-green-lightest`
					: tw`bg-gray-lightest`}
			}
		}
	`;

export const getStyles = ({ onClick, border, isSelected }: any) => {
	const styles = [baseStyle, getBorderStyles(border), getCursorStyles(onClick)];

	styles.push(getHoverStyles(isSelected));

	return styles;
};
