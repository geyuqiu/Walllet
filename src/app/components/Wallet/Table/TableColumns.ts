export const tableColumns = [
	{
		Header: 'Txid',
		accessor: 'id',
	},
	{
		Header: 'Sender',
		accessor: 'sender',
	},
	{
		Header: 'Recipient',
		accessor: 'recipient',
	},
	{
		Header: 'Timestamp',
		accessor: 'timestamp',
	},
	{
		Header: 'Amount',
		accessor: 'amount',
	},
	{
		Header: 'Fee',
		accessor: 'fee',
	}
];
