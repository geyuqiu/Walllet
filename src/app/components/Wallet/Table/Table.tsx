import React, {useMemo} from "react";
import {useTable} from 'react-table';
import styled from 'styled-components';

import {tableStyle} from './Table.styles';

const TableWrapper = styled.div`
	${tableStyle}
`;

export const Table = ({
	columns, transactions
}: TableProps) => {
	const tableData = useMemo(() => transactions, [transactions]);
	const tableColumns = useMemo(() => columns, [columns]);
	const {
		getTableProps, getTableBodyProps, headerGroups, rows, prepareRow
	} = useTable({
		data: tableData,
		columns: tableColumns
	});

	return (
		<TableWrapper>
			<table {...getTableProps()}>
				<thead>
				{headerGroups.map((headerGroup: any, headerGroupIndex: number) => (
					<tr {...headerGroup.getHeaderGroupProps()} key={headerGroupIndex}>
						{headerGroup.headers.map((column: any, thIndex: number) => (
							<th key={thIndex} {...column.getHeaderProps()}>{column.render('Header')}</th>
						))}
					</tr>
				))}
				</thead>
				<tbody {...getTableBodyProps()}>
				{rows.map((row, trIndex) => {
					prepareRow(row)
					return (
						<tr {...row.getRowProps()} key={trIndex}>
							{row.cells.map((cell, tdIndex) => <td {...cell.getCellProps()} key={tdIndex}>{cell.render('Cell')}</td>)}
						</tr>
					)
				})}
				</tbody>
			</table>
		</TableWrapper>
	);
};

type TableProps = {
	columns: any[];
	transactions: any[]
};

