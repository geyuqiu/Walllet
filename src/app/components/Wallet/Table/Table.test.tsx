import { render } from "@testing-library/react";
import React from "react";

import { Table } from "./Table";

describe("Table", () => {
	it("should match snapshot", () => {
		const { container } = render(<Table />);
		expect(container).toMatchSnapshot();
	});
});
