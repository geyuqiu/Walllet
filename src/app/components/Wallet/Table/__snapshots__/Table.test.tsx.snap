// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should match snapshot 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
